pkg/policies/branch/branch.go

```diff
@@ -267,6 +288,42 @@ func check(ctx context.Context, rep repositories, c *github.Client, owner,
 				d.BlockForce = false
 			}
 		}
+		rsc := p.GetRequiredStatusChecks()
+		if rsc != nil {
+			d.RequireUpToDateBranch = rsc.Strict
+			if mc.RequireUpToDateBranch && !rsc.Strict {
+				text = text +
+					fmt.Sprintf("Require up to date branch not configured for branch %v\n",
+						b)
+				pass = false
+			}
+			d.RequireStatusChecks = rsc.Contexts
+			c := make(map[string]struct{}, len(rsc.Contexts))
+			for _, ctx := range rsc.Contexts {
+				c[ctx] = struct{}{}
+			}
+			for _, check := range mc.RequireStatusChecks {
+				if _, ok := c[check]; !ok {
+					text = text +
+						fmt.Sprintf("Status check %v not required for branch %v\n",
+							check, b)
+					pass = false
+				}
+			}
+		} else {
+			d.RequireUpToDateBranch = false
+			if mc.RequireUpToDateBranch {
+				text = text +
+					fmt.Sprintf("Require up to date branch not configured for branch %v\n",
+						b)
+				pass = false
+			}
+			if len(mc.RequireStatusChecks) > 0 {
+				text = text +
+					fmt.Sprintf("No status checks required for branch %v\n", b)
```

Comment:
I might re-word this to: "Status checks required by policy, but none found."

Reply:
Updated in 754144bd3f9e24f3554793450990d44d4116b98f